### PR TITLE
feat: [CDS-37790]: Custom Approval step support

### PR DESCRIPTION
**Pull request by shamala-harness**
##### Summary:

Adds support for custom shell script approval step - Behind FF

##### Screenshots:

https://user-images.githubusercontent.com/96409598/171602852-58089db2-a895-4177-b001-9264fb8b2e1a.mov

Screenshot 2022-06-02 at 2 43 21 PM
Screenshot 2022-06-02 at 2 43 47 PM
Screenshot 2022-06-02 at 3 14 13 PM

Screenshot 2022-06-06 at 3 35 27 AM
Screenshot 2022-06-06 at 3 36 01 AM
Screenshot 2022-06-06 at 3 38 03 AM
Screenshot 2022-06-06 at 3 59 24 AM

Screenshot 2022-06-13 at 11 12 02 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
retrigger titlecheck

**Comment by samarth-sharma31** [comment]
dockerize

**Comment by samarth-sharma31** [comment]
dockerize

**Comment by samarth-sharma31** [comment]
dockerize

**Comment by samarth-sharma31** [comment]
retrigger jest

**Comment by samarth-sharma31** [comment]
dockerize

**Comment by samarth-sharma31** [comment]
dockerize

**Review by satyam-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/CustomApproval.tsx [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+      resolve([])
+    })
+  }
+
+  private getInitialValues(initialValues: CustomApprovalData): CustomApprovalFormData {
+    return {
+      ...initialValues,
+      spec: {
+        ...initialValues.spec,
+        shell: 'Bash',
```
Comment (devinder-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/CustomApproval.tsx:298]: can we move these to be `constants` in some util ?

**Review by devinder-harness — APPROVED** [review]

**Review by satyam-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/CustomApproval.tsx (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/CustomApproval.tsx:298]: sure, will do it in a follow up PR which is in progress - as i'll be adding more coverage there

**Pull request merged**